app/src/main/java/pl/edu/zut/mad/appzut/fragments/WebPlanFragment.java

```diff
@@ -0,0 +1,148 @@
+                web.loadUrl("javascript:javascript:(function(){document.body.appendChild(document.createElement('script')).src='/appwizut-injected-script.js'})()");
+                web.loadUrl("javascript:var x =document.getElementById('ctl00_ctl00_ContentPlaceHolder_MiddleContentPlaceHolder_txtIdent').value = '"+username+"';");
+                web.loadUrl("javascript:var y =document.getElementById('ctl00_ctl00_ContentPlaceHolder_MiddleContentPlaceHolder_txtHaslo').value = '"+password+"';");
+                web.loadUrl("javascript:var z =document.getElementById('ctl00_ctl00_ContentPlaceHolder_MiddleContentPlaceHolder_butLoguj').click();");
+                web.loadUrl("javascript:android.onData(document.getElementById('ctl00_ctl00_ContentPlaceHolder_MiddleContentPlaceHolder_lblMessage').innerHTML)");
+            }
+
+            @Override
+            @SuppressWarnings("deprecation") // Newer version is not supported on older Android versions
+            public WebResourceResponse shouldInterceptRequest(WebView view, String url) {
+                if (url.endsWith("/appwizut-injected-script.js")) {
+                    return new WebResourceResponse(
+                            "text/javascript",
+                            "utf-8",
+                            getResources().openRawResource(R.raw.grab_schedule)
+                    );
+                }
+                return null;
+            }
+
+            // Handle response from javascript
+            @Override
+            public boolean shouldOverrideUrlLoading(WebView view, String url) {
+                if (url.startsWith("js-grabbed-table:")) {
+                    String tableJson;
+                    try {
+                        tableJson = URLDecoder.decode(url.substring(17), "utf-8");
+                    } catch (UnsupportedEncodingException e) {
+                        throw new RuntimeException(e);
+                    }
+                    tableGrabbedByJavascript(tableJson);
+                    return true;
+                }
+                return false;
+            }
+
+            @Override
+            public void onPageStarted(WebView view, String url, Bitmap favicon) {
+
+            }
+
+            @Override
+            @SuppressWarnings("deprecation") // Newer version is not supported on older Android versions
+            public void onReceivedError(WebView view, int errorCode, String description, String failingUrl) {
+                mReceivedErrorForUrl = failingUrl;
+            }
+        });
+
+        web.loadUrl("https://edziekanat.zut.edu.pl/WU/PodzGodzin.aspx");
+    }
+
+    public void tableGrabbedByJavascript(final String contents) {
+        DataLoadingManager
+                .getInstance(getContext())
+                .getLoader(ScheduleEdzLoader.class)
+                .setSourceTableJson(contents);
+        getActivity().finish();
+    }
+
+    @JavascriptInterface
+    public void onData(String value) {
```

Comment:
This method needs better name (Because after all it's not handling data we came for here)